test/storage/sql_table_test.cpp

```diff
@@ -246,105 +316,677 @@ class SqlTableRW {
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row2.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  pc = pc_pair.first.Initialize(buffer);
+  // Need to scan again to get the rest of the data
+  table.table_->Scan(txn, &start_pos, pc, pc_pair.second, table.version_);
+  EXPECT_EQ(pc->NumTuples(), 2);
+  EXPECT_TRUE(start_pos == table.table_->end());
+
+  auto row3 = pc->InterpretAsRow(0);
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(0)));
+  EXPECT_NE(value, nullptr);
+  id = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_FALSE(seen_map[id]);
+  seen_map[id] = true;
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(1)));
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  // check the if we get (200, 10001, null)
+  auto row4 = pc->InterpretAsRow(1);
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(0)));
+  EXPECT_NE(value, nullptr);
+  id = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_FALSE(seen_map[id]);
+  seen_map[id] = true;
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(1)));
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  for (auto iter : seen_map) EXPECT_TRUE(iter.second);
+
+  delete[] buffer;
+  txn_manager_.Commit(txn, TestCallbacks::EmptyCallback, nullptr);
+  delete txn;
 }
 
 // NOLINTNEXTLINE
 TEST_F(SqlTableTests, VarlenInsertTest) {
-  SqlTableRW table(catalog::table_oid_t(2));
+  SqlTableTestRW table(catalog::table_oid_t(2));
+  auto txn = txn_manager_.BeginTransaction();
 
   table.DefineColumn("id", type::TypeId::INTEGER, false, catalog::col_oid_t(0));
   table.DefineColumn("datname", type::TypeId::VARCHAR, false, catalog::col_oid_t(1));
   table.Create();
 
-  table.StartRow();
-  table.SetIntColInRow(0, 100);
-  table.SetVarcharColInRow(1, "name");
-  storage::TupleSlot row_slot = table.EndRowAndInsert();
+  table.StartInsertRow();
+  table.SetIntColInRow(catalog::col_oid_t(0), 100);
+  table.SetVarcharColInRow(catalog::col_oid_t(1), "name");
```

Comment:
You should also use longer varlen values to test for varlen entries which are not inlined.